### ISSUE TITLE
triggers an action that is tracked for undo

### ISSUE THREAD

**Issue by dander401** [sompylasar/atom-cursor-indent#1]
with this package installed. cmd-z will undo cursor move movement rather than file editing

**Comment by sompylasar** [comment]
Yes, this is by design, the content of the file may change with each move. Atom does not support virtual whitespace for cursor movement without content change.

Any idea how the editor should behave, given this limitation?

**Comment by dander401** [comment]
I'm saying this not knowing anything about how atom works... so sorry if its nonsense.

Could it only make a content change on lines where it needs to insert whitespace? That way it at least would only cause the undos to happen when it went over those lines.

**Comment by sompylasar** [comment]
It should behave this way, it only changes the lines where the cursor moves from and to.

Perhaps this line should be guarded to avoid triggering mutations if nothing would actually change after them:
https://github.com/sompylasar/atom-cursor-indent/blob/49efa78668e2d5d9587695e432d629e0d8fb7610/lib/atom-cursor-indent.coffee#L31

I cannot do the same for the autoindent which is used to put the correct indentation onto the whitespace line:
https://github.com/sompylasar/atom-cursor-indent/blob/49efa78668e2d5d9587695e432d629e0d8fb7610/lib/atom-cursor-indent.coffee#L36
https://github.com/sompylasar/atom-cursor-indent/blob/49efa78668e2d5d9587695e432d629e0d8fb7610/lib/atom-cursor-indent.coffee#L57

Maybe there is another function to get the indentation but not apply the change, and this indentation may be compared to the current state of the buffer and avoid mutating if nothing would change.

**Comment by dander401** [comment]
i think we are saying the same thing. a change should only be triggered on lines where whitespace must be added.

**Comment by nmrugg** [comment]
For what it is worth, I removed the lines

```
  removeTrailingWhitespace(editor, currLineInfo)
  editor.groupChangesSinceCheckpoint(undoCheckpoint)
```

and now it only triggers an undo on lines where whitespace was added.

**Comment by sompylasar** [comment]
I found a flag that should skip the undo, https://github.com/atom/atom/blob/a79ffa308acf97375d0e4363aa4505216b004176/spec/text-editor-spec.coffee#L3009
Worth looking into passing it into whitespace changes.

**Comment by sompylasar** [comment]
I've published [v0.2.0](https://github.com/sompylasar/atom-cursor-indent/releases/tag/v0.2.0) that should fix the issue.

@dander401 @nmrugg I apologize for the delay. Please try the new version and share your experience, thanks!

**Comment by sompylasar** [comment]
On the [implementation](https://github.com/sompylasar/atom-cursor-indent/commit/e36136a1520ec3b9c45499f9e552ca1b0f02d4ef) with the [flag I found](https://github.com/sompylasar/atom-cursor-indent/issues/1#issuecomment-237953530): I had to copy the body of `editor.autoIndentSelectedRows` ([`autoIndentBufferRow`](https://github.com/atom/atom/blob/42509544b65472b7742c2ac34a2c88aa7e996617/src/text-editor.js#L4392-L4395) and [`setIndentationForBufferRow`](https://github.com/atom/atom/blob/42509544b65472b7742c2ac34a2c88aa7e996617/src/text-editor.js#L3485-L3488)) to provide `setTextInBufferRange` with the `{ undo: 'skip' }` option.
